src/main/kotlin/app/revanced/patches/all/connectivity/fakewifi/patch/FakeWifiPatch.kt

```diff
@@ -0,0 +1,252 @@
+            "registerNetworkCallback",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/app/PendingIntent;"),
+            "V",
+        ),
+        RegisterNetworkCallback3(
+            "Landroid/net/ConnectivityManager;",
+            "registerNetworkCallback",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;"),
+            "V",
+        ),
+        RequestNetwork1(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;"),
+            "V",
+        ),
+        RequestNetwork2(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "I"),
+            "V",
+        ),
+        RequestNetwork3(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;"),
+            "V",
+        ),
+        RequestNetwork4(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/app/PendingIntent;"),
+            "V",
+        ),
+        RequestNetwork5(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;", "I"),
+            "V",
+        );
+
+        fun replaceInstruction(method: MutableMethod, instruction: Instruction35c, instructionIndex: Int) {
+            val registers = arrayOf(instruction.registerC, instruction.registerD, instruction.registerE, instruction.registerF, instruction.registerG)
+            val argsNum = this.methodParams.size + 1 // + 1 for instance of definedClassName
+            if (argsNum > registers.size) {
+                // should never happen, but just to be sure (also for the future) a safety check
+                throw RuntimeException("not enough registers for ${this.definedClassName}#${this.methodName}: required $argsNum registers, but only got ${registers.size}")
+            }
+            val args = registers.take(argsNum).joinToString(separator = ", ") { reg -> "v${reg}" }
+            val replacementMethodDefinition = "${this.methodName}(${this.definedClassName}${this.methodParams.joinToString(separator = "")})${this.returnType}"
+            method.replaceInstruction(
+                instructionIndex,
+                "invoke-static { $args }, ${INTEGRATIONS_CLASS_DESCRIPTOR}->${replacementMethodDefinition}"
+            )
+        }
+
+        companion object {
+            fun fromMethodReference(methodReference: MethodReference) = values().firstOrNull { search ->
```

Comment:
You can use `MethodUtil.methodSignaturesMatch` here.

Reply:
using
```kotlin
MethodUtil.methodSignaturesMatch(methodReference, object : MethodReference {
    override fun getDefiningClass() = search.definedClassName
    override fun getName() = search.methodName
    override fun getParameterTypes() = search.methodParams.toList()
    override fun getReturnType() = search.returnType
    override fun validateReference() = Unit
    override fun compareTo(other: MethodReference) = throw NotImplementedError()
})
```
crashed my app for some reason.

Reply:
Which exception did you get? In any case this looks too complex, you can use the current implementation.

Reply:
[log.txt](https://github.com/revanced/revanced-patches/files/10473228/log.txt)